### PR TITLE
Structure CLI for Drone

### PR DESCRIPTION
**Pull request by yuzhouliu9**
Adds the basic CLI set up. A follow up PR will add in the flags/settings for `manifest` command.
The pluginType is passed in via ldflags which allows us to give drone or other tool specific set up.

### Testing
Push to local registry and run to see it works
```
make local-run
```

Pass in environment variables as drone would and see the take effect
```
> podman run -e PLUGIN_LOG_LEVEL=debug registry.example.com/buildah-plugin:latest
time="2023-07-21T17:35:29Z" level=info msg="To be implemented... log-level: debug"
```

### CODE REVIEW

File: internal/cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
+	log.Infof("To be implemented... log-level: %v", viper.GetString("log-level"))
+	return nil
+}
+
+func pluginTypeSetup() error {
+	pluginType := version.Get().PluginType
+	switch pluginType {
+	case version.PluginTypeDrone:
```
Comment (yuzhouliu9) [inline internal/cli/cli.go:65]: The plugin type is passed in via ldflags and drone specific set up is done here.
All drone plugin settings are passed in as environment variables with the prefix `PLUGIN_`.

**Comment by whwalter** [comment]
I am not sure we need to ship the drone docker runner binaries as part of this repo's test frame works. 

If you want to test the plugin locally you can run the container itself and pass the PLUGIN_ prefixed environment variables. This is a common pattern and you can see it documented in the [drone-kaniko repo](https://github.com/drone/drone-kaniko#manual-tagging). 

For end to end testing in a pipeline with a runner, we can add a multi-arch image build test to the main line in [drone-e2e-test](https://github.com/10gen/drone-e2e-test/blob/master/.drone.star) similar to the kaniko and paths changed tests.

**Comment by yuzhouliu9** [comment]
Gotcha, removed the drone docker runner binaries and test folder. Only have the relevant code changes now.

**Review by whwalter — APPROVED** [review]
LGTM

```
docker run --rm --env PLUGIN_LOG_LEVEL=debug registry.example.com/buildah-plugin:latest
time="2023-07-21T17:59:25Z" level=info msg="To be implemented... log-level: debug"
```

**Pull request merged**